src/RA_AchievementSet.cpp

```diff
@@ -17,36 +14,50 @@
 #include "services\ILeaderboardManager.hh"
 #include "services\ServiceLocator.hh"
 
+#include <array>
+#include <fstream>
+#include <memory>
+
 AchievementSet* g_pCoreAchievements = nullptr;
 AchievementSet* g_pUnofficialAchievements = nullptr;
 AchievementSet* g_pLocalAchievements = nullptr;
 
-AchievementSet** ACH_SETS[] ={ &g_pCoreAchievements, &g_pUnofficialAchievements, &g_pLocalAchievements };
-static_assert(SIZEOF_ARRAY(ACH_SETS) == NumAchievementSetTypes, "Must match!");
+inline constexpr std::array<AchievementSet**, 3> ACH_SETS
+{
+    &g_pCoreAchievements,
+    &g_pUnofficialAchievements,
+    &g_pLocalAchievements
+};
 
-AchievementSetType g_nActiveAchievementSet = Core;
+AchievementSet::Type g_nActiveAchievementSet = AchievementSet::Type::Core;
 AchievementSet* g_pActiveAchievements = g_pCoreAchievements;
 
-
-void RASetAchievementCollection(AchievementSetType Type)
+_Use_decl_annotations_
+void RASetAchievementCollection(AchievementSet::Type Type) noexcept
 {
     g_nActiveAchievementSet = Type;
-    g_pActiveAchievements = *ACH_SETS[Type];
+    g_pActiveAchievements = *ACH_SETS.at(ra::etoi(Type));
 }
 
-std::wstring AchievementSet::GetAchievementSetFilename(ra::GameID nGameID)
+_Use_decl_annotations_
+std::wstring AchievementSet::GetAchievementSetFilename(ra::GameID nGameID) noexcept
 {
+    std::wstring str;
+    {
+        std::wostringstream oss;
+        oss << g_sHomeDir << RA_DIR_DATA << nGameID;
+        str = oss.str();
+    }
     switch (m_nSetType)
     {
-        case Core:
-            return g_sHomeDir + RA_DIR_DATA + std::to_wstring(nGameID) + L".txt";
-        case Unofficial:
-            return g_sHomeDir + RA_DIR_DATA + std::to_wstring(nGameID) + L".txt";	// Same as Core
-        case Local:
-            return g_sHomeDir + RA_DIR_DATA + std::to_wstring(nGameID) + L"-User.txt";
-        default:
-            return L"";
+        case Type::Core:
+            return str += L".txt";
+        case Type::Unofficial:
+            return str += L".txt";	// Same as Core
+        case Type::Local:
+            return str += L"-User.txt";
```

Comment:
Speed things up a bit.

Reply:
might as well include this in the `oss` too:
```
    std::wstring str;
    {
        std::wostringstream oss;
        oss << g_sHomeDir << RA_DIR_DATA << nGameID;
        if (m_nSetType == Type::Local)
            oss << L"-User";
        oss << L".txt";
        str = oss.str();
    }
```

Reply:
Haven't thought of that, thanks.